lambda_function/src/file_processor/file_processor.py

```diff
@@ -194,35 +227,55 @@ def _process_file(self) -> None:
                     new_file_path = calibration.process_file(file_path)[0].name
 
                     # Get new file key
-                    new_file_key = self._generate_file_key(new_file_path)
+                    new_file_key = create_s3_file_key(
+                        science_filename_parser, new_file_path
+                    )
 
                     # Upload file to destination bucket if not a dry run
                     if not self.dry_run:
                         # Upload file to destination bucket
-                        self._upload_file(
-                            new_file_path, destination_bucket, new_file_key
+                        upload_file_to_s3(
+                            s3_client=self.s3_client,
+                            destination_bucket=destination_bucket,
+                            filename=new_file_path,
+                            file_key=new_file_key,
                         )
-
-                        if self.slack_client:
-                            # Send Slack Notification
-                            self._send_slack_notification(
-                                slack_client=self.slack_client,
-                                slack_channel=self.slack_channel,
-                                slack_message=(
-                                    f"File ({new_file_path}) has "
-                                    "been successfully processed and "
-                                    f"uploaded to {destination_bucket}.",
-                                ),
+                        try:
+                            if self.slack_client:
+                                # Send Slack Notification
+                                send_pipeline_notification(
+                                    slack_client=self.slack_client,
+                                    slack_channel=self.slack_channel,
+                                    path=new_file_path,
+                                    alert_type="processed",
+                                )
+                        except Exception as e:
+                            log.error(
+                                {
+                                    "status": "ERROR",
+                                    "message": f"Error when logging to Slack: {e}",
+                                }
```

Comment:
Do you need another try/catch here if there is already one in the utils function for `send_pipeline_notification`?

Reply:
Great point; I've fixed this and removed the redundant try and catches from this function.